src/cmd/cmd_wasm2wat.ml

```diff
@@ -4,13 +4,19 @@
 
 open Syntax
 
-let cmd_one file =
+(* Utility function to handle writing to a file or printing to stdout *)
+
+let cmd_one emitfile file =
   let ext = Fpath.get_ext file in
+  let wat_file = Fpath.set_ext "wat" file in
+
   match ext with
   | ".wasm" ->
     let* m = Parse.Binary.Module.from_file file in
     let m = Binary_to_text.modul m in
-    Ok (Fmt.pr "%a@\n" Text.pp_modul m)
+    let wat_module = Fmt.str "%a@\n" Text.pp_modul m in
+    if emitfile then Bos.OS.File.write wat_file wat_module
+    else Ok (Fmt.pr "%s" wat_module)
```

Comment:
It probably does not matter in practice, but it would be a little bit more efficient to avoid allocating a string:

```ocaml
let m = Binary_to_text.modul m in
if emitfile then Bos.OS.File.writef wat_file "%a@\n" Text.pp_modul m
else Fmt.str "%a@\n" Text.pp_modul m;
Ok ()
```

(It's untested so there may be a type error in the call to `writef` :sweat_smile:)

Reply:
Doing like this give an error,
```ocaml
else Fmt.str "%a@\n" Text.pp_modul m;
Ok ()
```

This works fine,
```ocaml
else Ok (Fmt.pr "%a@\n" Text.pp_modul m)
```

Reply:
Aw OK, I guess `Bos.OS.File.writef` already returns a result ? That's fine with me. :)

Reply:
@vasucp1207, please ping me once this is done so I can merge. Otherwise I'll forget.

Reply:
@zapashcanon, do you want me to make any changes🤔? It works correctly.

Reply:
Oh sorry, for some reasons I overlooked the changes. This is good to go now, thanks! :D